experimenter/experimenter/segments/__init__.py

```diff
@@ -7,6 +9,22 @@
 from experimenter.experiments.constants import NimbusConstants
 
 
+@dataclass
+class MockCommit:
+    committed_date: int
+
+
+@dataclass
+class MockRepository:
+    git_dir: Path
+
+    def iter_commits(self, *args, **kwargs):
+        mock_commit = MockCommit(
+            committed_date=int(datetime.now().timestamp()),
+        )
+        yield mock_commit
```

Comment:
I wish this wasn't necessary, but I do think this is a reasonable approach. A couple suggestions here though:

- can you rename the classes to indicate that they're private by prefixing with `_`?
- might be worth a comment to indicate how they're used? i.e., that they're used exclusively to allow the ConfigCollection to treat the local copy of metric-hub as a git repo, which is needed due to how `from_local_repo` works?

Reply:
We don't do `class _PrivateCls` anywhere else in the tree. I don't think we should start now.

```
~/W/s/g/m/experimenter $ rg "class _"
experimenter/typings/graphene/pyutils/signature.pyi
20:class _void:
25:class _empty: ...
27:class _ParameterKind(int):

experimenter/typings/graphene/utils/props.pyi
5:class _OldClass: ...
6:class _NewClass: ...

experimenter/typings/factory/declarations.pyi
63:class _UNSPECIFIED: ...
203:class _FactoryWrapper:

experimenter/typings/factory/base.pyi
98:class _Counter:

experimenter/typings/celery/_state.pyi
36:class _TLS(threading.local):

experimenter/typings/import_export/forms.pyi
31:    class _ExportActionForm(ActionForm):

experimenter/typings/celery/beat.pyi
218:class _Threaded(Thread):

experimenter/typings/celery/canvas.pyi
248:class _chain(Signature):
354:class _basemap(Signature):
462:class _chord(Signature):

experimenter/typings/celery/utils/functional.pyi
150:class _regen(UserList, list):

experimenter/typings/celery/utils/threads.pyi
69:class _LocalStack:
153:class _FastLocalStack(threading.local):

experimenter/typings/celery/utils/abstract.pyi
11:class _AbstractClass(metaclass=ABCMeta):

experimenter/typings/celery/bin/logtool.pyi
17:class _task_counts(list):

experimenter/typings/celery/utils/time.pyi
53:class _Zone:

experimenter/typings/celery/utils/deprecated.pyi
39:class _deprecated_property:
````

Reply:
Can I suggest we add something to metric_config_parser so that this isn't necessary? This seems like the wrong layer of abstraction.

Reply:
Yea.. I was willing to accept this as an alternative to creating new functionality just for this one use-case, but I think you're right. I'll go ahead and add something on the m-c-p side that we can call here without these mocks.

Reply:
I referenced this PR in the metric-config-parser PR, but just for completeness: https://github.com/mozilla/metric-hub/pull/683